packages/core/nuxt-module/src/modules/server-url.ts

```diff
@@ -0,0 +1,19 @@
+import consola from 'consola';
+
+export default function VueStorefrontServerUrl(): void {
+  if (!this.options?.publicRuntimeConfig?.middlewareUrl) {
+    this.nuxt.hook('listen', (server, { https, host, port }) => {
+      try {
+        if (!host || !port) throw new Error('Nuxt was\'t able to fetch your host or/and port.');
+        const baseURL = new URL('', `http${https ? 's' : ''}://${host}:${port}`);
```

Comment:
how about reverse-proxy? think about environment around the app - when you put load-balancer on front / VPC network / reverse proxy the app will receive a different address/domain than the one exposed, that may cause issues and break the app

Reply:
That's why you should set `middlewareUrl` with it.

Reply:
@andrzejewsky this solution is the same that we had in the past, just not using the `req` to fetch that data.

Reply:
.... and header.